feat(patterns/modals): add modal pattern

tiloyi opened the pull request:
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

Add Modal Pattern

GitHub issue number or Jira issue URL: N/A

## Other information

caqueste commented:
Deployment link → https://featmodaldev-dot-demo-dot-mozaic-design-system-dtrp.ey.r.appspot.com/Components/Modals/

nherwegh commented on src/docs/Components/Modals/code.mdx:
```diff
@@ -0,0 +1,116 @@
+
+In cases where it is not necessary, you can use a modal without heading, as follows:
+
+<Preview path="withouttitle" />
```
modals can't have two primary buttons

tiloyi replied:
My bad. It's fixed.

nherwegh commented on src/docs/Components/Modals/code.mdx:
```diff
@@ -0,0 +1,116 @@
+
+### Footer - two call to actions
+
+A primary call to action button accompanied with a secondary button to give the user an alternative
```
maybe finish the sentence with "alternative choice."

tiloyi replied:
fixed

nherwegh commented:
@tiloyi no "e" in every Modal Title preview 😉

tiloyi commented:
> @tiloyi no "e" in every Modal Title preview 😉

fixed ;)

nherwegh commented:
@tiloyi sorry about that but call to actions must take 100% of the modal's width on `breakpoint-s` as shown [here](https://share.goabstract.com/78533af2-8f38-4e31-b3d1-b6dbf88effb3?sectionId=273f281d-7e32-4043-87a6-88a0105abd30). I just fixed it after a feedback my bad, sorry!

tiloyi commented:
> @tiloyi sorry about that but call to actions must take 100% of the modal's width on `breakpoint-s` as shown [here](https://share.goabstract.com/78533af2-8f38-4e31-b3d1-b6dbf88effb3?sectionId=273f281d-7e32-4043-87a6-88a0105abd30). I just fixed it after a feedback my bad, sorry!

Fixed

nherwegh commented on src/docs/Components/Modals/code.mdx:
```diff
@@ -0,0 +1,143 @@
+
+A primary call to action button with a link to redirect the user to a page to give more informations.
+
+<Preview path="cta-link" />
```
the link should be centered on small breakpoint

tiloyi replied:
fixed

tiloyi commented:
> You can add a part Javascript needs and accessibility

fixed

tiloyi commented:
> @tiloyi The overlay between the page and the modal is missing.
> The properties are G900 with 70% of opacity with 1 of the background blur effect.

fixed but not for the "blur effect" because I don't have the value of this effect in the abstract file.

nherwegh commented:
> > @tiloyi The overlay between the page and the modal is missing.
> > The properties are G900 with 70% of opacity with 1 of the background blur effect.
> 
> fixed but not for the "blur effect" because I don't have the value of this effect in the abstract file.

maybe:
backdrop-filter: blur(1px);
